backend/fms_core/filters.py

```diff
@@ -75,6 +76,23 @@ def is_pooled_filter(self, queryset, name, values):
         bool_value = (values == 'true')
         return queryset.filter(is_pooled=bool_value)
 
+    def metadata_filter(self, queryset, name, values):
+        condition = Q()
+        # Metadata is of form: 'name1__value1, name2__value2, name2__Value3'
+        for metadatum in values.split(','):
+            if metadatum:
+                name_value = metadatum.split('__')
```

Comment:
There is still no validation in sample_metadata to prevent "__" in the name. You can add a new metadata_name_validator to prevent specifically that sequence of characters.